bookops_marc/bib.py

```diff
@@ -269,203 +236,137 @@ def main_entry(self) -> Optional[Field]:
+        else:
             return None
 
-    def remove_unsupported_subjects(self) -> None:
-        """
-        Deletes subject fields from the record that contain
-        unsupported by BPL or NYPL thesauri
-        """
-        subjects = self.subjects
-
-        for field in subjects:
-            if field.tag not in SUPPORTED_SUBJECT_TAGS:
-                self.remove_field(field)
-                continue
-            if field.indicator2 == "0":  # LCSH
-                continue
-            if field.indicator2 == "7":
-                if "2" in field:
-                    if field.get(code="2").strip() in SUPPORTED_THESAURI:
-                        continue
-                    else:
-                        self.remove_field(field)
-                else:
-                    self.remove_field(field)
-            else:
-                self.remove_field(field)
-
+    @property
     def physical_description(self) -> Optional[str]:
         """
         Returns value of the first 300 MARC tag in the bib
         """
         try:
-            return self.physicaldescription()[0].value()  # type: ignore
+            return self.physicaldescription[0].value()
         except (TypeError, IndexError):
             return None
 
+    @property
     def record_type(self) -> Optional[str]:
         """
         Retrieves record type code from MARC leader
         """
-        return self.leader[6]  # type: ignore
+        return self.leader[6]
 
+    @property
     def sierra_bib_format(self) -> Optional[str]:
         """
         Returns Sierra bib format fixed field code
         """
-        try:
-            return self.get("998").get(code="d").strip()  # type: ignore
-        except (TypeError, AttributeError):
+        field = self.get("998")
+        if not field:
+            return None
+        bib_format = field.get("d")
+        if isinstance(bib_format, str):
+            return bib_format.strip()
+        else:
             return None
 
+    @property
     def sierra_bib_id(self) -> Optional[str]:
         """
         Retrieves Sierra bib # from the 907 MARC tag
         """
-        try:
-            bib_id = self.get("907").get(code="a")[1:]  # type: ignore
-        except (TypeError, AttributeError):
+        field = self.get("907")
+        if not field:
             return None
-
-        if bib_id:
-            return bib_id  # type: ignore
+        bib_id = field.get("a")
+        if isinstance(bib_id, str) and len(bib_id) > 0:
+            return bib_id[1:]
         else:
             return None
 
+    @property
     def sierra_bib_id_normalized(self) -> Optional[str]:
         """
         Retrieves Sierra bib # from the 907 tag and returns it
         without 'b' prefix and the check digit.
         """
-        try:
-            return self.sierra_bib_id()[1:-1]  # type: ignore
-        except TypeError:
+        bib_id = self.sierra_bib_id
+        if isinstance(bib_id, str):
+            return bib_id[1:-1]
```

Comment:
Should we normalize Sierra bib ID as `int` to keep in line with order number (`Order.oid` returns an integer)?